src/datepicker/datepicker-input.ts

```diff
@@ -0,0 +1,204 @@
+  providers: [NGB_DATEPICKER_VALUE_ACCESSOR]
+})
+export class NgbInputDatepicker implements ControlValueAccessor {
+  private _cRef: ComponentRef<NgbDatepicker> = null;
+  private _model: NgbDate;
+  private _zoneSubscription: any;
+
+  /**
+   * Reference for the custom template for the day display
+   */
+  @Input() dayTemplate: TemplateRef<DayTemplateContext>;
+
+  /**
+   * First day of the week, 0=Sun, 1=Mon, etc.
+   */
+  @Input() firstDayOfWeek: number;
+
+  /**
+   * Callback to mark a given date as disabled
+   */
+  @Input() markDisabled: (date: {year: number, month: number, day: number}) => boolean;
+
+  /**
+   * Min date for the navigation. If not provided will be 10 years before today or `startDate`
+   */
+  @Input() minDate: {year: number, month: number, day: number};
+
+  /**
+   * Max date for the navigation. If not provided will be 10 years from today or `startDate`
+   */
+  @Input() maxDate: {year: number, month: number, day: number};
+
+  /**
+   * Whether to display navigation or not
+   */
+  @Input() showNavigation: boolean;
+
+  /**
+   * Whether to display days of the week
+   */
+  @Input() showWeekdays: boolean;
+
+  /**
+   * Whether to display week numbers
+   */
+  @Input() showWeekNumbers: boolean;
+
+  /**
+   * Date to open calendar with. If nothing provided, calendar will open with current month.
+   * Use 'navigateTo(date)' as an alternative
+   */
+  @Input() startDate: {year: number, month: number};
+
+  private _onChange = (_: any) => {};
+  private _onTouched = () => {};
+
+
+  constructor(
+      private _parserFormatter: NgbDateParserFormatter, private _elRef: ElementRef, private _vcRef: ViewContainerRef,
+      private _renderer: Renderer, private _cfr: ComponentFactoryResolver, ngZone: NgZone) {
+    this._zoneSubscription = ngZone.onStable.subscribe(() => {
+      if (this._cRef) {
+        positionElements(this._elRef.nativeElement, this._cRef.location.nativeElement, 'bottom-left');
+      }
+    });
+  }
+
+  /**
+   * @internal
+   */
+  registerOnChange(fn: (value: any) => any): void { this._onChange = fn; }
+
+  /**
+   * @internal
+   */
+  registerOnTouched(fn: () => any): void { this._onTouched = fn; }
+
+  /**
+   * @internal
+   */
+  writeValue(value) {
+    this._model = value ? new NgbDate(value.year, value.month, value.day) : null;
+    this._writeModelValue(this._model);
+  }
+
+  /**
+   * @internal
+   */
+  setDisabledState(isDisabled: boolean): void {
+    this._renderer.setElementProperty(this._elRef.nativeElement, 'disabled', isDisabled);
+    if (this.isOpen()) {
+      this._cRef.instance.setDisabledState(isDisabled);
+    }
+  }
+
+  /**
+   * @internal
+   */
+  manualDateChange(value: string) {
+    this._model = this._parserFormatter.parse(value);
+    this._onChange({year: this._model.year, month: this._model.month, day: this._model.day});
+    this._writeModelValue(this._model);
+  }
+
+  isOpen() { return !!this._cRef; }
```

Comment:
`isOpen()` is consistent with what we've got in other directives (dropdown, typeahead, ...). Please open a separate issue if you think that we should change it across the project.